src/next/UserError.ts

```diff
@@ -0,0 +1,7 @@
+export class UserError extends Error {
+    constructor(e: Error) {
+        super(e.message);
+        this.name = e.name;

```

Comment:
What do these two lines do?

Reply:
I find the following definition, but the constructor accepts `message?:string` as the only parameter. I add these lines to copy the values.
```typescript
interface Error {
    message: string;
    name?: string;
    stack?: string;
}
```

Reply:
I agree these lines look strange. I think following image can explain the reason:
![image](https://user-images.githubusercontent.com/2351748/45073515-30241980-b112-11e8-9fcf-da6f9d88737d.png)

What if users want to preserve some extra information in the original error? Is it right to simply discard them when casting it into UserError?